tests/test_confluence_api.py

```diff
@@ -60,74 +86,74 @@ def test__configure_atlassian(self, confluence_client):
             == new_sess.adapters["https://"].total
         )
 
-    def test_find_code_metadata(self, confluence_client):
-        """Test find_code_metadata method with mocked response."""
-
-        def mock_response(url):
-            """Return a mock response object with code block element."""
-
-            class MockResponse:
-                @property
-                def content(self):
-                    if url == "https://example.com/no_code_block":
-                        return b"<div>No code block here</div>"
-                    elif url == "https://example.com/single_code_block":
-                        return b'<code>{"title": "awesome project"}</code>'
-                    elif url == "https://example.com/multiple_code_blocks":
-                        return (
-                            b'<code>{"title": "first project"}</code>'
-                            + b'<code>{"title": "second project"}</code>'
-                        )
-
-            return MockResponse()
-
+    def test_extract_json_metadata(self, confluence_client, response_fixt):
+        """Test extract_json_metadata method with mocked response."""
         # set up
         client = confluence_client
         # single code block should pass -----------------------------------
         # Mock the _get_atlassian_page_content method
-        url = "https://example.com/single_code_block"
+        url = "https://example.com/json_code_block"
         when(client)._get_atlassian_page_content(url).thenReturn(
-            mock_response
+            response_fixt
         )
-        client.response = mock_response(url)
+        client.response = response_fixt(url)
         # Use and assert
-        metadata = client.find_code_metadata(url)
+        metadata = client.extract_json_metadata(url)
         # Assert the expected metadata
         assert isinstance(metadata, dict)
-        expected_metadata = {"title": "awesome project"}
+        expected_metadata = {"title": "json project"}
         assert metadata == expected_metadata
         unstub()
         # no code block must raise ----------------------------------------
         url = "https://example.com/no_code_block"
         when(client)._get_atlassian_page_content(url).thenReturn(
-            mock_response
+            response_fixt
         )
-        client.response = mock_response(url)
+        client.response = response_fixt(url)
         with pytest.raises(
             ValueError, match="No code elements were found on this page."
```

Comment:
I'd probably put this on the previous line, as it's a bit hard to read like this.